packages/stores/src/schema-store/actions/execute-operation.ts

```diff
@@ -19,6 +31,46 @@ import type {
   SchemaStoreActions,
 } from "../schema-store.types";
 
+enum Directive {
+  Defer = "defer",
+}
+
+const usingDefer = (set: SelectionSetNode | undefined): boolean =>
+  set?.selections.some((selection) => {
+    // add kind.FRAGMENT_SPREAD once we support named fragments
+    const inlineFragment = selection.kind == Kind.INLINE_FRAGMENT;
+    const usingDeferOnCurrentSelection =
+      inlineFragment &&
+      (selection.directives?.some(
+        (node) => node.name.value === Directive.Defer
+      ) ??
+        false);
+    const hasSelectionSet = "selectionSet" in selection;
+    return (
+      usingDeferOnCurrentSelection ||
+      (hasSelectionSet && usingDefer(selection.selectionSet))
+    );
+  }) ?? false;
+
+const mergeResults = (
+  result: ExecutionResult<Record<string, unknown>, unknown>,
+  lastResult?: ExecutionResult<Record<string, unknown>, unknown>
+) => {
+  // bit of weird typing here, result should probably be returned as ExecutionResult | ExecutionPatchResult from graphql-sse
+  // but that isn't the case
+  if (!("path" in result) || lastResult == null) {
+    return result;
+  }
+
+  const path = result.path as string[];
+
+  const combined = set(lastResult, ["data", ...path], result.data);
```

Comment:
I don't think `set` is the right approach.  If the query looks like 

```
query {
   someField
   ... @defer {
     someOtherField
   }
}
```

Then we're going to get an initial response with `someField` in it, then a subsequent response with a path of `[]`. Unless I'm misunderstanding this set would overwrite the original `someField`? 

That was why I suggested an updateIn & merge combo in slack

Reply:
This overwrites the original, `lastResult` is the former response and not the current one

Reply:
You're right, will swap this.

Reply:
If we run that query I provided we'll get two responses:

```
{"data": {"someField": hello"}}

{"data": {"someOtherField": "blah"}, "path": []}
```

So we'll end up doing `set({"data": {"someField": "hello"}}, ["data"], {"someOtherField": "blah"})` right?

And the output of that is going to be `{"data": {"someOtherField": "blah"}}`.  `someField` has been overwritten, because you did a set not a merge.

Reply:
Fixed